Note Preview - Crash if editing a card in a Dynamic Deck

Thanks to anonymous beta tester f9852c31-9c99-4617-9d72-91e65ac40ef9 

https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/f7d8edbb-1005-4a25-98e7-bfa10702e548

```
java.lang.RuntimeException: Unable to resume activity {com.ichi2.anki/com.ichi2.anki.CardTemplatePreviewer}: com.ichi2.utils.JSONException: org.json.JSONException: No value for maxTaken
at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4430)
at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4470)
at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:51)
at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2199)
at android.os.Handler.dispatchMessage(Handler.java:112)
at android.os.Looper.loop(Looper.java:216)
at android.app.ActivityThread.main(ActivityThread.java:7625)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:987)
Caused by: com.ichi2.utils.JSONException: org.json.JSONException: No value for maxTaken
at com.ichi2.utils.JSONObject.get(JSONObject.java:2)
at org.json.JSONObject.getInt(JSONObject.java:481)
at com.ichi2.utils.JSONObject.getInt(JSONObject.java:1)
at com.ichi2.libanki.Card.timeLimit(Card.java:2)
at com.ichi2.libanki.Card.timeTaken(Card.java:2)
at com.ichi2.anki.AbstractFlashcardViewer.resumeTimer(AbstractFlashcardViewer.java:3)
at com.ichi2.anki.AbstractFlashcardViewer.onResume(AbstractFlashcardViewer.java:2)
at com.ichi2.anki.CardTemplatePreviewer.onResume(CardTemplatePreviewer.java:1)
at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1456)
at android.app.Activity.performResume(Activity.java:7614)
at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4412)
... 11 more
Caused by: org.json.JSONException: No value for maxTaken
at org.json.JSONObject.get(JSONObject.java:392)
at com.ichi2.utils.JSONObject.get(JSONObject.java:1)
... 21 more
```

--- a/AnkiDroid/src/main/java/com/ichi2/anki/CardTemplatePreviewer.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/CardTemplatePreviewer.java
@@ -170,7 +170,11 @@ public class CardTemplatePreviewer extends AbstractFlashcardViewer {
         }
 
         if (mNoteEditorBundle != null) {
-            mCurrentCard.setDid(mNoteEditorBundle.getLong("did"));
+            long newDid = mNoteEditorBundle.getLong("did");
+            if (col.getDecks().isDyn(newDid)) {
+                mCurrentCard.setODid(mCurrentCard.getDid());
+            }
+            mCurrentCard.setDid(newDid);
 
             Note currentNote = mCurrentCard.note();
             ArrayList<String> tagsList = mNoteEditorBundle.getStringArrayList("tags");

